src/Event/ReadModel/History/EventPlaceHistoryProjector.php

```diff
@@ -0,0 +1,166 @@
+use CultuurNet\UDB3\EventHandling\DelegateEventHandlingToSpecificMethodTrait;
+use CultuurNet\UDB3\Model\Place\PlaceIDParser;
+use CultuurNet\UDB3\Model\ValueObject\Identity\UUID;
+use CultuurNet\UDB3\Model\ValueObject\Web\Url;
+use CultuurNet\UDB3\ReadModel\DocumentDoesNotExist;
+use CultuurNet\UDB3\ReadModel\DocumentRepository;
+use Psr\Log\LoggerInterface;
+
+class EventPlaceHistoryProjector implements EventListener
+{
+    use DelegateEventHandlingToSpecificMethodTrait;
+
+    private EventLocationHistoryRepository $repository;
+    private DocumentRepository $eventRepository;
+    private LoggerInterface $logger;
+
+    public function __construct(
+        EventLocationHistoryRepository $eventLocationHistoryRepository,
+        DocumentRepository $eventRepository,
+        LoggerInterface $logger
+    ) {
+        $this->repository = $eventLocationHistoryRepository;
+        $this->eventRepository = $eventRepository;
+        $this->logger = $logger;
+    }
+
+    protected function applyLocationUpdated(LocationUpdated $event): void
+    {
+        try {
+            $oldPlaceId = $this->getOldPlaceUuid($event->getItemId());
+        } catch (DocumentDoesNotExist $e) {
+            $this->logger->error(sprintf('Failed to store location updated: %s', $e->getMessage()));
+            return;
+        }
+
+        $this->repository->storeEventLocationMove(
+            new UUID($event->getItemId()),
+            $oldPlaceId,
+            new UUID($event->getLocationId()->toString())
+        );
+    }
+
+    protected function applyEventCreated(EventCreated $event): void
+    {
+        $this->repository->storeEventLocationStartingPoint(
+            new UUID($event->getEventId()),
+            new UUID($event->getLocation()->toString())
+        );
+    }
+
+    protected function applyEventCopied(EventCopied $event): void
+    {
+        try {
+            $placeId = $this->getOldPlaceUuid($event->getOriginalEventId());
+        } catch (DocumentDoesNotExist $e) {
+            $this->logger->error(sprintf('Failed to store event copied: %s', $e->getMessage()));
+            return;
+        }
+
+        $this->repository->storeEventLocationStartingPoint(
+            new UUID($event->getItemId()),
+            $placeId
+        );
+    }
+
+    protected function applyEventImportedFromUDB2(EventImportedFromUDB2 $event): void
+    {
+        try {
+            $udb2Event = EventItemFactory::createEventFromCdbXml(
+                $event->getCdbXmlNamespaceUri(),
+                $event->getCdbXml()
+            );
+        } catch (CultureFeed_Cdb_ParseException $e) {
+            $this->logger->error(sprintf('Failed to store event imported from UDB2: %s', $e->getMessage()));
+            return;
+        }
+
+        $this->repository->storeEventLocationStartingPoint(
+            new UUID($event->getEventId()),
+            new UUID($udb2Event->getLocation()->getCdbid())
```

Comment:
I think we need to take into account that a lot of older places have an XML without a Cdbid, this should be caught and not make the handler crash.